src/figuresClasses.ts

```diff
@@ -1,19 +1,92 @@
+export enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
+}
+
+export type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
 export class Triangle {
+  shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public sideA: number,
+    public sideB: number,
+    public sideC: number,
+  ) {
+    if (this.sideA <= 0 || this.sideB <= 0 || this.sideC <= 0) {
+      throw new Error(
+        'Every side of the triangle cannot be 0 or a negative number',
+      );
+    }
+
+    if (
+      this.sideA + this.sideB <= this.sideC
+      || this.sideA + this.sideC <= this.sideB
+      || this.sideB + this.sideC <= this.sideA
+    ) {
+      throw new Error(
+        'The longest side of triangle should be less than a sum of two others',
+      );
+    }
```

Comment:
looks good. Would be even better if you have it separated in the method like `validateTriangle`